src/localization/gps_linearization.py

```diff
@@ -31,15 +33,45 @@ def __init__(self):
         self.ref_alt = rospy.get_param("gps_linearization/reference_point_altitude")
 
         self.world_frame = rospy.get_param("gps_linearization/world_frame")
+        self.odom_frame = rospy.get_param("gps_linearization/odom_frame")
         self.rover_frame = rospy.get_param("gps_linearization/rover_frame")
 
+    def publish_pose(self):
+        """
+        Publishes the pose of the rover in relative to the map frame to the TF tree,
+        either as a direct map->base_link transform, or if the odom frame is in use,
+        indirectly as a map->odom transform.
+        """
+        rover_in_map = self.pose
+
+        try:
```

Comment:
yea just the one thing about the configuration based switch, maybe we still need exception handling but most of it should be based on an if